### PR TITLE
Export low level API used by NAS Home UI as TigerSpecificFunctions

### PR DESCRIPTION
**Pull request by xMort**
JIRA: FET-855

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8336/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10208/ : FAILURE in 8m 06s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8340/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10214/ : SUCCESS in 9m 04s

**Review by kandl — APPROVED** [review]
LGTM

File: libs/sdk-backend-tiger/src/backend/index.ts [hunk 1]
```diff
@@ -193,8 +308,312 @@ export class TigerBackend implements IAnalyticalBackend {
...
+                            const result = await sdk.entities.getEntityOrganizations({ id: organizationId });
+                            return result.data?.data?.attributes?.allowedOrigins || [];
+                        });
+                    } catch (error) {
+                        return [];
+                    }
+                },
+                getOrganizationPermissions: async (organizationId: string) => {
+                    return await this.authApiCall(async (sdk) => {
+                        try {
```
Comment (no23reason) [inline libs/sdk-backend-tiger/src/backend/index.ts:323]: Should the style be unified? `getOrganizationAllowedOrigins` has try catch on top of authApiCall, `getOrganizationPermissions` has it inverted. I am not sure it is intentional, if not we should pick one approach and stick with it.

File: libs/sdk-backend-tiger/src/backend/index.ts [hunk 2]
```diff
@@ -193,8 +308,312 @@ export class TigerBackend implements IAnalyticalBackend {
...
+                            const result = await sdk.entities.createEntityDataSources({
+                                jsonApiDataSourceInDocument: sampleDataSource
+                                    ? { ...sampleDataSource }
+                                    : {
+                                          data: {
+                                              attributes: {
+                                                  name: "demo-ds",
+                                                  url: "jdbc:postgresql://localhost:5432/demo",
```
Comment (no23reason) [inline libs/sdk-backend-tiger/src/backend/index.ts:501]: Do we need this default? Should this always come from outside instead? I am a bit "triggered" by the connection string hardcoded here.

File: libs/sdk-backend-tiger/src/backend/index.ts [hunk 3]
```diff
@@ -193,8 +308,312 @@ export class TigerBackend implements IAnalyticalBackend {
...
+                                    },
+                                },
+                            });
+                            return result.data.data.id;
+                        } catch (error) {
+                            throw convertApiError(error);
+                        }
+                    });
+                },
+                createWorkspaceFromDocument: async (workspace: JsonApiWorkspaceInDocument) => {
```
Comment (no23reason) [inline libs/sdk-backend-tiger/src/backend/index.ts:547]: Can we maybe join `createWorkspace` and `createWorkspaceFromDocument` they seem relatively similar.

File: libs/sdk-backend-tiger/src/backend/index.ts (hunk 2 above)
Comment (xMort) [reply libs/sdk-backend-tiger/src/backend/index.ts]: Yeah, I was not sure about this either. I removed that default and use it directly from where it is called. It makes more sense now 👍

Comment (xMort) [reply libs/sdk-backend-tiger/src/backend/index.ts]: Yeah, I was not sure about this either. I removed that default and use it directly from where it is called. It makes more sense now 👍

File: libs/sdk-backend-tiger/src/backend/index.ts (hunk 3 above)
Comment (xMort) [reply libs/sdk-backend-tiger/src/backend/index.ts]: I did not liked this one either. I dig more from where the document is provided (it's `gdc-public-demos` repo from where it is fetched via API call 🥲). I reused the `createWorkspace` function and deleted this one.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8342/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10218/ : FAILURE in 8m 11s

**Review by no23reason — APPROVED** [review]
LGTM

File: libs/sdk-backend-tiger/src/backend/index.ts (hunk 3 above)
Comment (no23reason) [reply libs/sdk-backend-tiger/src/backend/index.ts]: Thanks :)

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2407/ : SUCCESS in 3m 16s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2702/ : SUCCESS in 25s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1520/ : SUCCESS in 28m 08s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10220/ : SUCCESS in 13m 41s

**Pull request merged**